backend/src/main/java/gov/cdc/usds/simplereport/api/apiuser/ApiUserMutationResolver.java

```diff
@@ -33,27 +35,52 @@ public User addUser(
             String email,
             String organizationExternalID
                 ) {
-        _us.assertEmailAvailable(email);
         ApiUser apiUser = _us.createUser(email, firstName, middleName, lastName, suffix, organizationExternalID);
-        Optional<OrganizationRoles> orgRoles = _os.getOrganizationRolesForUser(apiUser);
+        Optional<OrganizationRoles> orgRoles = _us.getOrganizationRolesForUser(apiUser.getInternalId());
+        Boolean isAdmin = _us.isAdmin(apiUser);
+        return new User(apiUser, orgRoles, isAdmin);
+    }
+    
+    public User addUserToCurrentOrg(
+            String firstName,
+            String middleName,
+            String lastName,
+            String suffix,
+            String email
+                ) {
+        ApiUser apiUser = _us.createUserInCurrentOrg(email, firstName, middleName, lastName, suffix);
+        Optional<OrganizationRoles> orgRoles = _us.getOrganizationRolesForUser(apiUser.getInternalId());
```

Comment:
if you're okay with me moving the `User` (api model) class down to the `ApiUserService` level, then I can do this no problem. The issue is that I want "create a user" to be atomic with "add that user to the specified Okta org" inside `createUserInCurrentOrg()`, but the return value of that method right now is limited to `ApiUser`, which is a DB entity that doesn't capture Okta relationships.

Reply:
I think the proper solution would be to more or less move `User` to the `service.model` package, thus resolving the package dependency cycle. (And also to implement #663, to fix the behavior split weirdness.) Absent that change, I think I'd rather not push `User` into the `ApiUserService` interface, for the reasons you mention. 

That makes this feel like a particular type of refactor, which is the one where I give myself a 15-30 minute window to try to implement it, and if it keeps getting more complicated I do a hard reset, sigh heavily, and move on with my day. So for now, I would say either try creating a service model that looks a whole lot like `User` and see how much of a PITA it would be to use it, or just punt and link this discussion from #663 (where it is already presumably linked in any case) as an additional corner to be filed off when we pull in that work.

Reply:
well, be careful what you wish for, but i did it :)

Reply:
(moving `User`, that is. I'm leaving https://github.com/CDCgov/prime-simplereport/issues/663 for another PR.)